back/app/models/initiative.rb

```diff
@@ -113,7 +113,15 @@ class Initiative < ApplicationRecord
   def cosponsor_ids=(ids)
     return unless ids
 
-    super(ids.uniq)
+    ids = ids.uniq
+    current_ids = cosponsors.pluck(:id).uniq
+    return if current_ids.sort == ids.sort
+
+    cosponsors_initiatives.where.not(user_id: ids).destroy_all
+
+    (ids - current_ids).each do |id|
+      cosponsors_initiatives.create(user_id: id)
+    end
```

Comment:
I reverted to my more verbose solution, since we need to use `destroy` to get around the error:
```Ruby
PG::ForeignKeyViolation: ERROR:  update or delete on table "cosponsors_initiatives" violates foreign key constraint "fk_rails_828a073a04" on table "notifications"
```
... when an associated `notification` exists.

Conventionally, we get around this by using the `before_destroy :remove_notifications` hook that I have also added to `back/app/models/cosponsors_initiative.rb`